indexing-service/src/main/java/io/druid/indexing/overlord/MetadataTaskStorage.java

```diff
@@ -244,6 +244,75 @@ public void addLock(final String taskid, final TaskLock taskLock)
   }
 
   @Override
+  public void upgradeLock(String taskid, TaskLock taskLockToUpgrade)
+  {
+    Preconditions.checkNotNull(taskid, "taskid");
```

Comment:
Same comment as above- I think this should throw an IllegalArgumentException if `taskLockToUpgrade` is preemptible -- callers should only be providing non-preemptible locks to this method.